src/main/java/com/jfrog/ide/idea/scan/ScannerBase.java

```diff
@@ -272,6 +269,10 @@ private void addImpactPathToDependencyNode(DependencyNode dependencyNode, List<S
             if (currImpactTreeNode == null) {
                 currImpactTreeNode = new ImpactTreeNode(currPathNode);
                 parentImpactTreeNode.getChildren().add(currImpactTreeNode);
+                if (pathNodeIndex == path.size() - 1) {
+                    // If a new leaf was added, thus a new impact path was added (impact paths don't collide after they split)
+                    impactTree.incImpactPathsCount();
```

Comment:
Why did we move `impactTree.incImpactPathsCount()` from line 263?

Reply:
Before the change, every call to addImpactPathToDependencyNode() incremented the count also if affectively no path was added (while using Yarn why command sometimes the same path can be extracted). This change ensures that the counter is updated while a new path was actually added.The code is based on the fact that every new path ends with a new leaf in our impact tree.